js/index.ts

```diff
@@ -19,67 +21,62 @@ app.all('/', urlencodedParser, async (req, res, next) => {
 
     const {host} = req.headers;
 
-    if(host === ROOT) {
-        // 
-        if(req.method === 'GET') {
-            res.sendFile(join(__dirname, '/../www/landing.html'))
-            return;
-        }
-
-        if(req.method === 'POST') {
-
-            const {subdomain} = req.body
+    if (host !== ROOT) {
+        next();
+        return;
+    }
 
-            if(valid(subdomain)) {
-                if(await store.exists({subdomain})) {
+    if(req.method === 'GET') {
+        res.sendFile(join(__dirname, '/../www/landing.html'))
+        return;
+    }
 
-                    res.redirect(subdomain + ROOT)
-    
-                    return;
-                } else {
+    if (req.method === 'POST') {
 
-                    await store.createStore({subdomain})
+        const {subdomain} = req.body
 
-                    res.send(`CREATED SUBDOMAIN: ${req.body.subdomain}`)
-                    return;
-                }
+        if (valid(subdomain)) {
+            if (!await store.exists({subdomain})) {
+                await store.createStore({subdomain})
             }
 
-
-            return;
+            res.redirect(`http://${subdomain}.${ROOT}`)
+        } else {
+            res.status(404).send(`${subdomain} is not a valid subdomain name!`);
         }
 
+        return;
     }
-
-    next();
-
 })
 
 app.all('*', async (req, res) => {
     const {host} = req.headers;
 
-    const match =host?.match(SUBREG);
-    if(match) {
-
-        const {subdomain} = (match.groups as any)
+    const match = host?.match(SUBREG);
```

Comment:
I always knew seeing my first optional chaining in the wild would be special...but this is BEYOND!!!
🤩🤩🤩

Reply:
...I also obviously was not paying attention during the initial coding session on Saturday